File: sdk/go/stub_builder.go
package mockforge

// StubBuilder provides a fluent interface for creating response stubs
type StubBuilder struct {
	method    string
	path      string
	status    int
	headers   map[string]string
	body      interface{}
	latencyMs *int
}

// NewStubBuilder creates a new StubBuilder
func NewStubBuilder(method, path string) *StubBuilder {
	return &StubBuilder{
		method:  method,
		path:    path,
		status:  200,
		headers: make(map[string]string),
	}
}

// Status sets the HTTP status code
func (b *StubBuilder) Status(code int) *StubBuilder {
	b.status = code
	return b
}

// Header sets a response header
func (b *StubBuilder) Header(key, value string) *StubBuilder {
	b.headers[key] = value
	return b
}

// Headers sets multiple response headers
func (b *StubBuilder) Headers(headers map[string]string) *StubBuilder {
	for k, v := range headers {
		b.headers[k] = v
	}
	return b
}

// Body sets the response body
func (b *StubBuilder) Body(body interface{}) *StubBuilder {
	b.body = body
	return b
}

// Latency sets the response latency in milliseconds
func (b *StubBuilder) Latency(ms int) *StubBuilder {
	b.latencyMs = &ms
	return b
}

// Build builds the ResponseStub
func (b *StubBuilder) Build() ResponseStub {
	return ResponseStub{
		Method:    b.method,
		Path:      b.path,
		Status:    b.status,
		Headers:   b.headers,
		Body:      b.body,
		LatencyMs: b.latencyMs,
	}
}

File: sdk/nodejs/src/stubBuilder.ts
import { ResponseStub } from './types';

/**
 * Fluent builder for creating response stubs
 */
export class StubBuilder {
  private stub: Partial<ResponseStub>;

  constructor(method: string, path: string) {
    this.stub = {
      method: method.toUpperCase(),
      path,
      status: 200,
      headers: {},
    };
  }

  /**
   * Set the response status code
   */
  status(code: number): this {
    this.stub.status = code;
    return this;
  }

  /**
   * Set a response header
   */
  header(key: string, value: string): this {
    if (!this.stub.headers) {
      this.stub.headers = {};
    }
    this.stub.headers[key] = value;
    return this;
  }

  /**
   * Set multiple response headers
   */
  headers(headers: Record<string, string>): this {
    this.stub.headers = { ...this.stub.headers, ...headers };
    return this;
  }

  /**
   * Set the response body
   */
  body(body: any): this {
    this.stub.body = body;
    return this;
  }

  /**
   * Set response latency in milliseconds
   */
  latency(ms: number): this {
    this.stub.latencyMs = ms;
    return this;
  }

  /**
   * Build the response stub
   */
  build(): ResponseStub {
    if (!this.stub.body) {
      throw new Error('Response body is required');
    }
    return this.stub as ResponseStub;
  }
}
